tools/edge_cut/src/surf/meshUtils.cpp

```diff
@@ -0,0 +1,293 @@
+
+edge_cut::surf::K::FT
+edge_cut::surf::SizingField::operator()(  Point_3 const & p,
+                                          int     const,
+                                          Index   const &   ) const
+{
+  FT l_distance = std::sqrt( std::pow( p.x()-m_center.x(), 2 ) + std::pow( p.y()-m_center.y(), 2 ) );
+
+  if ( l_distance <= m_innerRad )
+    return m_innerVal;
+  else if ( l_distance >= m_outerRad )
+    return m_scale * m_innerVal;
+  else {
+    // Control should never reach here if inner and outer radii are equal
+    assert( m_outerRad != m_innerRad );
+    return ( 1 + ( l_distance - m_innerRad ) * ( m_scale - 1) * m_widthInv ) * m_innerVal;
+  }
+}
+
+
+edge_cut::surf::Polyline_type
+edge_cut::surf::topoIntersect(  Poly_slicer & i_slicer,
+                                K::Plane_3    i_plane   )
+{
+  Polylines     polylines;
+  Polyline_type ridges;
+
+  i_slicer( i_plane, std::back_inserter( polylines ) );
+
+  if( polylines.size() == 1 ) {
+    ridges = *(polylines.begin());
+  }
+  else {
+    EDGE_LOG_ERROR << "Error: Polyline slicer did not return precisely one intersection!";
+  }
+  return ridges;
+}
+
+
+bool
+edge_cut::surf::checkMonotonic( Polyline_type & i_p,
+                                unsigned int    i_n,
+                                bool            i_inc )
+{
+  if ( i_n > 2 ) {
+    EDGE_LOG_ERROR << "checkMonotonic:";
+    EDGE_LOG_ERROR << "  Input index is greater than 2";
+    return false;
+  }
+
+  for ( std::size_t l_idx = 1; l_idx < i_p.size(); l_idx++ ) {
+    if ( i_inc ) {
+      // Check monotonically increasing
+      if ( i_p[ l_idx - 1 ][ i_n ] >= i_p[ l_idx ][ i_n ] ) return false;
+    }
+    else {
+      // Check monotonically decreasing
+      if ( i_p[ l_idx - 1 ][ i_n ] <= i_p[ l_idx ][ i_n ] ) return false;
+    }
+  }
+
+  return true;
+}
+
+
+void
+edge_cut::surf::orderPolyline(  Polyline_type & i_p,
+                                unsigned int    i_n )
+{
+  // i_p is already monotonically increasing, so return
+  if ( checkMonotonic( i_p, i_n, true) )
+    return;
+  // i_p is monotonically decreasing, so reverse order of all points
+  else if ( checkMonotonic( i_p, i_n, false ) ) {
+    Polyline_type l_temp = i_p;
+    for ( std::size_t l_idx = 0; l_idx < i_p.size(); l_idx++ ) {
+      i_p[ l_idx ] = l_temp[ i_p.size() - 1 - l_idx ];
+    }
+  }
+  // i_p is not monotonic in the (i_n+1)th coordinate, record an error
+  else {
+    EDGE_LOG_ERROR << "orderPolyline:";
+    EDGE_LOG_ERROR << "  encountered a non-monotonic polyline - printing polyline:";
+    for ( auto const & l_pt : i_p )
+      EDGE_LOG_ERROR << "  " << l_pt;
+  }
+
+  return;
+}
+
+
+std::list< edge_cut::surf::Polyline_type >
+edge_cut::surf::getIntersectionFeatures(  Polyhedron  const & i_topoSurface,
+                                          double      const * i_bBox        )
+{
+  std::list< Polyline_type > features;
+
+  EDGE_LOG_INFO << "Computing topography-boundary intersection...";
```

Comment:
Don't think we should have an info-message here (silent functions).
Move to message outside.